src/go/rpk/pkg/kafka/client_test.go

```diff
@@ -0,0 +1,59 @@
+package kafka_test
+
+import (
+	"testing"
+
+	"github.com/Shopify/sarama"
+	"github.com/stretchr/testify/assert"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+)
+
+func Test_LoadConfig(t *testing.T) {
+	getCfg := func() *config.Config {
+		cfg := config.Default()
+		cfg.Rpk.SCRAM.User = "some_user"
+		cfg.Rpk.SCRAM.Password = "some_password"
+		cfg.Rpk.SCRAM.Type = sarama.SASLTypeSCRAMSHA256
+		return cfg
+	}
+	t.Run("when scram is provided", func(t *testing.T) {
+		cfg := getCfg()
+
+		c, err := kafka.LoadConfig(cfg)
+		assert.NoError(t, err)
+		assert.Equal(t, cfg.Rpk.SCRAM.User, c.Net.SASL.User)
+		assert.Equal(t, cfg.Rpk.SCRAM.Password, c.Net.SASL.Password)
+		assert.Equal(t, sarama.SASLMechanism(cfg.Rpk.SCRAM.Type), c.Net.SASL.Mechanism)
+	})
```

Comment:
I don't wanna delay this being merged, but I think we should stick to a single style for the tests. These could be expressed as a list like most of the other tests in the rpk code. Maybe something like this could work:

```go
tests := []struct{
    name string
    conf func() *config.Config
    check func(*testing.T, *saslconfig)
}{
    // test cases...
}
for _, tt := range tests {
    t.Run(tt.name, func(st *testing.T) {
        conf := getCfg()
        if (tt.conf != nil) {
                conf = tt.conf()
        }
        c, err := kafka.LoadConfig(conf)
        require.NoError(t, err)
        check(st, c)
    })
}
```

Reply:
Another change I would suggest is completing the description of the test case, like "when x, then y", so that it's easier to see what they test without reading each assertion.

Reply:
@0x5d do you want to commit this suggestion to this branch for rafal?

Reply:
For sure. I'll do that in a moment.

Reply:
I did. @RafalKorepta please check it out and let me know if there's anything wrong 🙏🏽

Reply:
LGTM